frontend/ui/elements/menu_keyboard_layout.lua

```diff
@@ -26,7 +34,21 @@ local sub_item_table = {
         text_func = function()
             local activated_keyboards = getActivatedKeyboards()
             if activated_keyboards ~= "" then
-                return string.format(_("Keyboard layout: %s"), activated_keyboards)
+                local item_text = string.format(_("Keyboard layout: %s"), activated_keyboards)
+
+                -- get width of text
+                local tmp = TextWidget:new{
+                    text = item_text,
+                    face = Font:getFace("cfont"),
+                }
+                local item_text_w = tmp:getSize().w
+                tmp:free()
+                if item_text_w >= Screen:getWidth() then
```

Comment:
https://github.com/koreader/koreader/blob/6d40c7c7ea8abe2e12e9897ee7d6ed0dbc5c8712/frontend/ui/widget/touchmenu.lua#L87-L89

Reply:
Yes I saw the line. But I have tested different keyboards and I never got an overflow (...) . Maybe the above line is a bit to restrictive?
Could you please play a bit with the given solution?

With the above line I get:
![grafik](https://user-images.githubusercontent.com/36999612/142038232-fc08ad96-5308-471b-9f27-01ac69b89c3e.png)
 and 
![grafik](https://user-images.githubusercontent.com/36999612/142038312-77c6a4f3-e889-4e6f-b25f-1348732bf592.png)

But as you see in the first picture: Ellenica (sorry for spelling) should easily fit, without truncation :(

Reply:
I would have stayed with the 2-char lang tag: changeing what is displayed depending if it's truncated feels a bit odd.
If it's because there is no 2-char lang tag in the menu content, well, I would add them, then just like they are in the KeyboardLayoutDialog (long-pres son the globe key).

Reply:
![2](https://user-images.githubusercontent.com/62179190/142041118-bc8cf0f2-7612-4cdf-8a47-5a05da3ab4e6.png)

Reply:
> If it's because there is no 2-char lang tag in the menu content, well, I would add them

They look rather good: https://github.com/koreader/koreader/pull/8142#issuecomment-907663703

Reply:
![3](https://user-images.githubusercontent.com/62179190/142042681-4f2d06b3-f9bf-4c66-91a9-945d3146a6dd.png)

I'm sure that in translation (to Russian) this will be truncated too.

Reply:
Truncation is a problem for Arabic languages -- the word being truncated needs to be written differently at the truncation point due to the rules of the writing system -- and in some cases "truncation" actually will result in the string being lengthened. The most famous example of this [was the "effective power" text message which could crash iPhones a few years ago](https://www.youtube.com/watch?v=hJLMSllzoLA), likely due to the truncation of Arabic characters. In addition, truncating using bytes is not ideal for the vast majority of unicode text because you could cut the text in the middle of a character (corrupting the last character) or cut it between combining characters which should not be split (the rules for whether you can split them or not depends on the language).

It's probably better to avoid this problem entirely imho.